servers/rendering/renderer_rd/renderer_scene_render_rd.cpp

```diff
@@ -567,7 +569,12 @@ void RendererSceneRenderRD::_render_buffers_post_process_and_tonemap(const Rende
 			// If we do a bilinear upscale we just render into our render target and our shader will upscale automatically.
 			// Target size in this case is lying as we never get our real target size communicated.
 			// Bit nasty but...
-			dest_fb = texture_storage->render_target_get_rd_framebuffer(render_target);
+
+			if (dest_is_msaa_2d) {
```

Comment:
@Calinou this fix might explain why things go wrong when 2D MSAA is turned on, it wasn't copying into the 2D MSAA buffer.

Haven't fully tested this yet but might be worth looking into further.

Reply:
Ok still looking into this, something still weird

Reply:
Ok, works correctly now, we somehow had lost the code that if scaling is 1.0 we ignore the scaling mode, so we were applying FSR even if there was no scaling to be done.

FSR has a problem that it can't render into our MSAA 2D buffer, so if MSAA 2D is enabled it had to copy stuff over. Extra step, sucks, but it's what it is.

Anyway, along side fixing the stereo issue, this now also fixes a couple of MSAA issues.